app/src/main/java/org/p2p/wallet/transaction/progresshandler/TransactionProgressHandler.kt

```diff
@@ -28,58 +43,92 @@ abstract class TransactionProgressHandler(private val glideManager: GlideManager
     protected lateinit var progressStateFormat: String
 
+    val context: Context
+        get() = binding.context
+
+    private val cellAdapter = CommonAnyCellAdapter(
+        mainCellDelegate()
+    )
+
     fun init(viewBinding: DialogNewTransactionProgressBinding, data: NewShowProgress) {
         binding = viewBinding
         progressStateFormat = viewBinding.getString(R.string.transaction_progress_title)
         handleInitState(data)
     }
 
-    open fun handleInitState(data: NewShowProgress) {
+    open fun handleInitState(showProgressData: NewShowProgress) {
         with(binding) {
-            val colorMountain = getColor(R.color.text_mountain)
             textViewSubtitle.text = getString(
                 R.string.transaction_date_format,
-                dateFormat.format(data.date),
-                timeFormat.format(data.date)
+                dateFormat.format(showProgressData.date),
+                timeFormat.format(showProgressData.date)
             )
-            glideManager.load(imageViewToken, data.tokenUrl, IMAGE_SIZE)
-            val amountInUsd = data.amountUsd
+            glideManager.load(imageViewToken, showProgressData.tokenUrl, IMAGE_SIZE)
+            val amountInUsd = showProgressData.amountUsd
             if (amountInUsd != null) {
-                textViewAmountUsd.text = data.amountUsd
-                textViewAmountTokens.text = data.amountTokens
+                textViewAmountUsd.text = showProgressData.amountUsd
+                textViewAmountTokens.text = showProgressData.amountTokens
             } else {
-                textViewAmountUsd.text = data.amountTokens
+                textViewAmountUsd.text = showProgressData.amountTokens
                 textViewAmountTokens.isVisible = false
             }
-            data.amountColor?.let { amountColorRes ->
+            showProgressData.amountColor?.let { amountColorRes ->
                 textViewAmountUsd.setTextColorRes(amountColorRes)
             }
-            if (data.recipient == null) {
-                textViewSendToTitle.isVisible = false
-                textViewSendToValue.isVisible = false
-            } else {
-                textViewSendToValue.text = data.recipient
+
+            recyclerViewTransactionData.apply {
+                attachAdapter(cellAdapter)
+                layoutManager = LinearLayoutManager(context)
             }
-            val totalFees = data.totalFees
-            textViewFeeValue.text = if (totalFees != null) {
-                buildSpannedString {
-                    totalFees.forEach { textToHighlight ->
-                        append(
-                            SpanUtils.highlightText(
-                                commonText = textToHighlight.commonText,
-                                highlightedText = textToHighlight.highlightedText,
-                                color = colorMountain
-                            )
+            val feesCellModel = mapTransactionDataCell(
```

Comment:
TransactionProgressHandler.kt is like a mapper itselfe it will be not good to override TransactionProgressHandler for needed stransaction types + some mappers
i mean it is base implementation
mappers better to add in overriden one if needed